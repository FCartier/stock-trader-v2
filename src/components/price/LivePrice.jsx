import React from "react";

const dynamicDiv = price => {
  let text = '';

  price > 0 
  ? text = "text-green"
  : text = "text-orange";

  return text;
}

const LivePrice = ({price}) => {
  const { lastSalePrice } = price;
  console.log(lastSalePrice)
  return Object.keys(price).length ? (
    <div>

        <div className="price">
            <sup>$</sup>{lastSalePrice}
        </div>

    </div>
  ) : null
};

export default LivePrice;