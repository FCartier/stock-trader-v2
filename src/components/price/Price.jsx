import React from "react";

const Price = ({price}) => {
   return renderCondition(price) ? (
    <div>
        <div className="price">
            <sup>$</sup>{price}
        </div>
    </div>
   ) : null
};


function renderCondition(price){
  if (price === 0) {
    return false
  } else {
    return true
  }
}


export default Price;