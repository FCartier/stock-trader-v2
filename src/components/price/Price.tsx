import * as React from "react";

export interface IPriceProps {
  price: {
    close: number;
    change: number;
    changePercent: number;
    lastSalePrice: number;
    askPrice: number;
  };
}

const Price: React.SFC<IPriceProps> = ({ price }) => {
  return price.close ? (
    <div>
      <div className="price">
        <sup>$</sup>
        {price.close.toFixed(2)}{" "}
        <div className={dynamicDiv(price.change)}>
          {price.change} | {price.changePercent}
          <sup>%</sup>
        </div>
      </div>
    </div>
  ) : (
    <div className="price">
      
      {price.askPrice ? (
        <div>
          <sup>$</sup>
          {price.askPrice.toFixed(2)}
        </div>
      ) : null} 
    </div>
  );
};

const dynamicDiv = (price: number) => {
  return price > 0 ? "text-green" : "text-orange";
};


export default Price;
