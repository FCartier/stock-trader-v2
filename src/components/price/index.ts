import PriceContainer from "./priceContainer"

export default PriceContainer