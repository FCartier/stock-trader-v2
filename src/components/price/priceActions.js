export const FETCH_PRICE_SUCCESS = "FETCH_PRICE_SUCCESS";
export const RECEIVED_NEW_PRICE = 'RECEIVED_NEW_PRICE'

export const priceReceived = payload => ({
    type: RECEIVED_NEW_PRICE,
    payload
  });
  
