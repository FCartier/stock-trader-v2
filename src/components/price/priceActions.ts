import { Action, ActionCreator } from "redux";

export const FETCH_STATIC_PRICE = "FETCH_STATIC_PRICE";
export type FETCH_STATIC_PRICE = typeof FETCH_STATIC_PRICE;

export const RECEIVED_NEW_PRICE = "RECEIVED_NEW_PRICE";
export type RECEIVED_NEW_PRICE = typeof RECEIVED_NEW_PRICE;

export interface StaticPriceAction extends Action {
  type: typeof FETCH_STATIC_PRICE
  payload: string
};

export interface PriceAction extends Action {
  type: typeof RECEIVED_NEW_PRICE;
  payload: string;
}

export const priceReceived: ActionCreator<PriceAction> = payload => ({
  type: RECEIVED_NEW_PRICE,
  payload
});

export const fetchStaticPrice: ActionCreator<StaticPriceAction> = payload => ({
  type: FETCH_STATIC_PRICE,
  payload
});
