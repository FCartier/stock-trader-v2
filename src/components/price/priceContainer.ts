import Price from "./Price";
import { connect } from "react-redux";
import { GlobalState } from "../../StoreTypes";

const mapStateToProps = ({ price, keystats: { keystats } }: GlobalState) => ({
  price,
  keystats
});

export default connect(
  mapStateToProps,
  null
)(Price);
