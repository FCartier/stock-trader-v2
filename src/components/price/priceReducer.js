import {FETCH_PRICE_SUCCESS} from "./priceActions"

export const price = (state = {}, action) => {
    switch (action.type) {
      case FETCH_PRICE_SUCCESS:
        return action.payload;
      default:
        return state;
    }
  };