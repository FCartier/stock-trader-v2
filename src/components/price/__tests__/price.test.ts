import { configure, shallow } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import Price from '../Price';
import { testData } from '../__data__/price.data';

configure({ adapter: new Adapter() })

describe("<Price />", () => {
  test('should render properly', () => {
    const component = shallow(Price({ price: testData }));
    expect(component).toMatchSnapshot();
  });
})