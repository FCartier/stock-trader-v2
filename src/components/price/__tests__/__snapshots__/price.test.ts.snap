// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Price /> should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div>
    <div
      className="price"
    >
      <sup>
        $
      </sup>
      221.19
       
      <div
        className="text-orange"
      >
        -0.96
         | 
        -0.432
        <sup>
          %
        </sup>
      </div>
    </div>
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="price"
      >
        <sup>
          $
        </sup>
        221.19
         
        <div
          className="text-orange"
        >
          -0.96
           | 
          -0.432
          <sup>
            %
          </sup>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <sup>
            $
          </sup>,
          "221.19",
          " ",
          <div
            className="text-orange"
          >
            -0.96
             | 
            -0.432
            <sup>
              %
            </sup>
          </div>,
        ],
        "className": "price",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "$",
          },
          "ref": null,
          "rendered": "$",
          "type": "sup",
        },
        "221.19",
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              -0.96,
              " | ",
              -0.432,
              <sup>
                %
              </sup>,
            ],
            "className": "text-orange",
          },
          "ref": null,
          "rendered": Array [
            -0.96,
            " | ",
            -0.432,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "%",
              },
              "ref": null,
              "rendered": "%",
              "type": "sup",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="price"
        >
          <sup>
            $
          </sup>
          221.19
           
          <div
            className="text-orange"
          >
            -0.96
             | 
            -0.432
            <sup>
              %
            </sup>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <sup>
              $
            </sup>,
            "221.19",
            " ",
            <div
              className="text-orange"
            >
              -0.96
               | 
              -0.432
              <sup>
                %
              </sup>
            </div>,
          ],
          "className": "price",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "$",
            },
            "ref": null,
            "rendered": "$",
            "type": "sup",
          },
          "221.19",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                -0.96,
                " | ",
                -0.432,
                <sup>
                  %
                </sup>,
              ],
              "className": "text-orange",
            },
            "ref": null,
            "rendered": Array [
              -0.96,
              " | ",
              -0.432,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "%",
                },
                "ref": null,
                "rendered": "%",
                "type": "sup",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
